### ISSUE TITLE
[Cpasbien] :: Request failed: 403 Client Error: Forbidden

### ISSUE THREAD

**Issue by snigel** [SickRage/SickRage#2932]
I'm getting the following repeatedly in my logs:
INFO     SEARCHQUEUE-BACKLOG-80467 :: [Cpasbien] :: Request failed: 403 Client Error: Forbidden for url: http://www.cpasbien.cm/top-100.php

1. Obviously this is due to the whole site showing 403 errors, even with regular browser.
What confuses me is that this is not treated the same way as HTTP errors with TorrentProject for example, see #2929 and #2930 should HTTP errors be counted as INFO or ERROR?

2. There is no handling for when this happens. Instead of switching to another provider or using a timeout on repeated HTTP errors, Sickrage keeps spamming the provider with more requests.

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by miigotu** [comment]
The fact that you are getting errors with multiple providers should show you that your settings are wrong.  SR can't handle if you set it up wrong.

**Comment by snigel** [comment]
How can a HTTP 403 be an error in my configuration, when I get the same HTTP error in the browser?
Can you please elaborate, it doesn't make sense to me.

My setup is mostly default, haven't changed search intervals. Configured where to store files, what torrent client to use and what providers to use.

My issue with the errors are how they are handled. Closing this ticket with the motivation that it's my fault that the errors are showing up is not changing the issues I'm seeing.

1. This HTTP error is INFO while TorrentProject throws ERROR for HTTP errors.
I haven't found a setting for that, nor have I memory of changing it.

2. My second issue is that SickRage keeps hammering a provider with new requests even though it's not available due to errors.
I haven't found a setting for that, nor have I memory of changing it.

If you honestly believe this is user error, please give me any pointers to what settings could have done this.

**Comment by miigotu** [comment]
In settings, advanced tab, cpu throttling set it to high.

**Comment by snigel** [comment]
There seem to be an misunderstanding.

The problem is that SR keeps sending requests even though the provider is returning errors.
Changing the search interval does not change that.

If search 1, 2 and 3 fails due to HTTP 403, there's no point in doing 20 more requests to that provider right now.

**Comment by miigotu** [comment]
Dude, it's not a bug, what don't you get?  Just because you don't like the answer doesn't mean the answer is incorrect or that you need to open multiple threads for similar questions. No-one is obligated to even answer you, let alone give you the answer you want or do the work you expect.

You are getting 403 now because that provider has hit you with a temp ban. The others give 503, or some other error. These are not errors, just status codes.
This is caused by SETTINGS in sickrage being wrong. Either cpu throttling too low (the delay between requests), daily or backlog search interval too low, or incorrect qualities set for shows.  If a show doesnt get released in some quality but your SR is constantly searching and spamming a provider for that quality for 100's of episodes they will block you. Sites don't like SR, get used to it.

**Comment by snigel** [comment]
It seems like you really want to believe that I'm a stupid user, expecting somebody else to fix issues that are due to my configuration. I'm truly sorry if that's how my tickets were perceived.

I'm solely discussing how errors are handled in SR. Not why they are appearing or what they mean.

>Dude, it's not a bug, what don't you get?
What is not a bug? I've been trying to pinpoint issues that I see, for example;

1. Handling the same type of errors in different ways sounds like a bug or at least inconsistent behavior to me. Counting one HTTP error as INFO and another as ERROR is not predictable behavior.

2. Hammering a provider (regardless of hammer rate) that we know is down, is not effective for getting files or for removing that ban. Perhaps it's not a bug, but it's certainly something that would be beneficial to improve.

Surely, you as a developer can say that this is the intended behavior and that's how you want it.
And that is perfectly fine.

> Just because you don't like the answer doesn't mean the answer is incorrect
I could reply exactly the same, just because you don't like the issue I see, doesn't mean they are non-issues.

> or that you need to open multiple threads for similar questions
In my experience multiple bugs/issues in the same ticket is frowned upon. Since I got three different (but similar) issues with three different providers, I thought that was the right thing to do. If it was not, I apologize.

> No-one is obligated to even answer you, let alone give you the answer you want or do the work you 
> expect.
Nobody is obligated to post issues they see. I don't expect anybody to do the work for me, I'm trying to help out by posting inconsistencies and issues that I see.

Turn things around for a while, you're not obligated to close tickets without understanding or explaining.
Doing so does not encourage me, or somebody else, to do a pull request.

I don't ask you to fix my configuration, I don't ask you to implement what I'm seeing.

I'm simply posting issues I'm seeing hoping that it will be either leaved open, acknowledged or closed for [reason]. Reasons could be "no timeouts is intended behavior", "Timeouts are too complex", "The SR team have no resources to ever implement this, please create a pull request, this will be closed".

>You are getting 403 now because that provider has hit you with a temp ban. The others give 503, or >some other error. These are not errors, just status codes.
Yes, as I've replied several times, I get what the errors mean and why I get them. It's the *handling* of the errors I've created issues for. I do not ask for help with why I'm getting these errors. I do not ask for help with what HTTP codes mean. It's the way they are handled in SR.

>This is caused by SETTINGS in sickrage being wrong. 
Again, I'm not asking why I'm getting a 403, I'm posting about *how* the 403 is handled.

>Either cpu throttling too low (the delay between requests), daily or backlog search interval too low, or >incorrect qualities set for shows. If a show doesnt get released in some quality but your SR is constantly >searching and spamming a provider for that quality for 100's of episodes they will block you. 
Even with the delay between requests turned up, it is futile to keep sending requests to a provider that is returning errors. Again, it's this behavior I'm reporting, not that I'm getting errors.

But what if the provider is actually down due to whatever reasons. Doesn't it make sense to put that provider in quarantine for X minutes and switch to another? It would certainly remove my ban faster and give me torrents from other providers faster, instead of clogging up the error log.

>Sites don't like SR, get used to it.
Wouldn't one reason to dislike SR as a siteop, be that SR keeps on hammering my site, regardless of what error codes the site sends back? It would certainly frustrate me, to have clients not backing off or using timeouts, even on repeated errors.

With all this said, if you think 403 should be INFO while 429 and 503 are ERROR it's your choice as a developer.
If you think the best way to handle HTTP errors from a provider is to keep on using that provider as if nothing has happened, that is again your choice as a developer.

But saying that INFO vs ERROR choice or SRs handling of said errors, are a configuration issue is simply not true.

**Comment by miigotu** [comment]
403 Is INFO because it is not caused by an error in our post data, the others can be caused by errors in post data.  The difference between error and info in SR is used for whether the issue submitter reports them or not automatically and we dont want issues that are not coding errors submitted to the issue submitter.

Also, we WANT similar issues in the same thread. Closed/Open does not matter here since we get notifications regardless, we just want them grouped by similar issues so we can see how common an issue is and/or possible ways to improve it easier.

If you had posted with a request, rather than making it out like SR was failing somehow this all would have went different. "[FR] Skip provider for X minutes after a failure" rather than opening multiple issues about http errors might have gotten you a better response.  I have thousands of hours, hundreds of thousands of changed lines, and thousand+ commits in this repo, so spamming me insinuating that I did something wrong without even making a request is insulting.

**Comment by snigel** [comment]
I had no intention of insulting the effort you spent or the code you wrote. I'm not insinuating anything, it's unlucky if it looks that way. I appreciate that people spend their personal time to write software that I can use for free. I write software for a living, so I get the effort you're spending here.

What I don't need to insinuate is that I felt insulted by you making this personal, treating me as a spoiled moron rather than actually reading what I wrote. I do find it insulted to get tickets closed without explanation, because I spent time trying to read debug logs, pinpoint what is happening, when it's happening before I was posting.

Regarding grouping, I tried to read the guide for writing tickets before posting, but the link is going back [here] (https://github.com/SickRage/SickRage/issues), to the list of open issues. So  I wrote tickets in the way that is generally accepted within my domain. One specific issue per ticket.

Nor did I did I try to shoehorn FRs as issues, I thought what I was posting was issues. For FRs I posted mine at [feathub](http://feathub.com/SickRage/SickRage), but not before reading through all the open ones, not to repost and to vote.
